opendatahub-io/odh-dashboard
# Pipeline experiments

closes: https://github.com/opendatahub-io/odh-dashboard/issues/1115

## Description


**Triggered with default experiment**
<img width="1134" alt="image" src="https://github.com/opendatahub-io/odh-dashboard/assets/12587674/fb1829db-0ce8-49d2-b87d-0938b3052734">

**empty experiment**
<img width="1128" alt="image" src="https://github.com/opendatahub-io/odh-dashboard/assets/12587674/c4b87527-6ad4-4397-a60b-23f8306a15a4">

**experiment with runs**
<img width="1113" alt="image" src="https://github.com/opendatahub-io/odh-dashboard/assets/12587674/4f057714-fcf6-49b4-9fd6-7bd540ff06b2">

**delete modal**
<img width="591" alt="image" src="https://github.com/opendatahub-io/odh-dashboard/assets/12587674/c976b97b-84af-4046-b52b-dae7af42e1ac">

## How Has This Been Tested?



1. create a new pipeline server
2. create a pipeline
3. create a experiment
4. go to pipeline runs experiments tab and check empty state
5. create a pipeline run and set the experiment to the one you created
6. check that the pipeline run is populated under the correct experiment
7. delete the experiment
8. experiment should still be visible because the pipeline is still linked, but you can not create any runs on that experiment or delete it again 
9. clone the pipeline you created and it should default to default

## Test Impact


pipelines tests are in issues

cc @kywalker-rh 

## Request review criteria:
